### PR TITLE
feat(editor): add refactoring.nvim

### PR DESCRIPTION
**Pull request by rubiin**
More on:
https://github.com/ThePrimeagen/refactoring.nvim#features

### CODE REVIEW

File: lua/lazyvim/plugins/extras/editor/refactoring.lua [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        c = false,
+        h = false,
+        hpp = false,
+        cxx = false,
+      },
+      printf_statements = {},
+      print_var_statements = {},
+    },
+    config = function(_, options)
+      require("refactoring").setup(options)
```
Comment (dpetka2001) [inline lua/lazyvim/plugins/extras/editor/refactoring.lua:117]:
```suggestion
    config = function(_, opts)
      require("refactoring").setup(opts)
```

File: lua/lazyvim/plugins/extras/editor/refactoring.lua [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      require("telescope").load_extension("refactoring")
```
Comment (dpetka2001) [inline lua/lazyvim/plugins/extras/editor/refactoring.lua:118]:
```suggestion
      if LazyVim.has("telescope.nvim") then
	      LazyVim.on_load("telescope.nvim", function()	
		      require("telescope").load_extension("refactoring")
		  end)
	  end
```

**Comment by dpetka2001** [comment]
@rubiin I made the suggestions on the github site and thus when I pressed `<Tab>` it indented with 4 spaces instead of 2. Maybe you would like to do the indentation with 2 spaces for conformity.

**Comment by rubiin** [comment]
> @rubiin I made the suggestions on the github site and thus when I pressed `<Tab>` it indented with 4 spaces instead of 2. Maybe you would like to do the indentation with 2 spaces for conformity. Sorry about that.

fixed

**Comment by folke** [comment]
Thanks!

**Pull request merged**